RealmSwift/PersistedProperty.swift

```diff
@@ -159,10 +159,31 @@ public struct Persisted<Value: _Persistable> {
         case let .unmanaged(value, _, _):
             return value
         case .unmanagedNoDefault:
-            let value = Value._rlmDefaultValue()
+            let value = Value._rlmDefaultValue(false)
             storage = .unmanaged(value: value)
             return value
-        case let .unmanagedObserved(value, _):
+        case let .unmanagedObserved(value, key):
+            var name: String = ""
+            if Value._rlmType == .linkingObjects {
+                name = RLMObjectBaseObjectSchema(object)!.computedProperties[Int(key)].name
+            } else {
+                name = RLMObjectBaseObjectSchema(object)!.properties[Int(key)].name
+            }
+            if object.lastAccessedNames != nil {
```

Comment:
Looking up the name should be inside this block rather than doing it even if we're not recording the keypath.